pkg/odo/cli/component/common_link.go

```diff
@@ -38,6 +58,55 @@ func (o *commonLinkOptions) complete(name string, cmd *cobra.Command, args []str
 
 	suppliedName := args[0]
 	o.suppliedName = suppliedName
+
+	if experimental.IsExperimentalModeEnabled() {
+		o.Context = genericclioptions.NewDevfileContext(cmd)
+
+		o.serviceType, o.serviceName, err = svc.IsOperatorServiceNameValid(suppliedName)
+		if err != nil {
+			return err
+		}
+
+		componentName := o.EnvSpecificInfo.GetName()
+
+		// Assign static/hardcoded values to SBR
+		o.sbr.Kind = sbrKind
+		o.sbr.APIVersion = strings.Join([]string{sbrGroup, sbrVersion}, "/")
+
+		// service binding request name will be like <component-name>-<service-type>-<service-name>. For example: nodejs-etcdcluster-example
+		o.sbr.Name = strings.Join([]string{componentName, strings.ToLower(o.serviceType), o.serviceName}, "-")
+		o.sbr.Namespace = o.EnvSpecificInfo.GetNamespace()
+		o.sbr.Spec.DetectBindingResources = true // because we want the operator what to bind from the service
```

Comment:
can we have a helper function called `generateSBR` which gets assigned to `o.sbr`?

Reply:
We make changes to `o.sbr` at different places in `complete`, `validate` and `run` based on the info we have at each stage and what makes sense where. A general helper might not be so helpful, IMO.